src/mediapipe_internal/mediapipegraphdefinition.cpp

```diff
@@ -100,6 +100,11 @@ Status MediapipeGraphDefinition::dryInitializeTest() {
 Status MediapipeGraphDefinition::validate(ModelManager& manager) {
     SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Started validation of mediapipe: {}", getName());
     ValidationResultNotifier notifier(this->status, this->loadedNotify);
+    auto& models = manager.getModels();
+    if (std::find_if(models.begin(), models.end(), [this](auto pair) { return this->getName() == pair.first; }) != models.end() || manager.pipelineDefinitionExists(this->getName())) {
```

Comment:
```suggestion
    if (manager.modelExists(this->getName()) || manager.pipelineDefinitionExists(this->getName())) {
```

Reply:
This way we leave thread safety as the responsibility of modelmanager. In fact we shouldn't expose getModels as public